src/eos-metrics-instrumentation.c

```diff
@@ -75,6 +86,94 @@ static gint64 start_time;
 
 static EinsPersistentTally *persistent_tally;
 
+static gboolean
+get_os_version (gchar **name_out,
+                gchar **version_out)
+{
+    g_autoptr(GError) error = NULL;
```

Comment:
I am hesitant to introduce auto pointers since they aren't used anywhere else in the system.